spec/models/miq_provision/state_machine_spec_helper.rb

```diff
@@ -0,0 +1,72 @@
+module MiqProvision::StateMachineSpecHelper
+  include MiqRequestTaskHelper
+
+  def test_prepare_provision
+    call_method
+  end
+
+  def test_poll_destination_in_vmdb
+    @test_poll_destination_in_vmdb_setup ||= begin
+      expect(task).to receive(:requeue_phase).twice { requeue_phase }
+      expect(task).to receive(:find_destination_in_vmdb).and_return(nil, nil, vm)
+    end
+    expect(task.destination).to be_nil
+
+    call_method
+  end
+
+  def test_poll_destination_powered_off_in_provider
+    expect(task).to receive(:powered_off_in_provider?).and_return(true, false, false, true)
+    test_poll_destination_powered_off_in_provider_no_callback
+    test_poll_destination_powered_off_in_provider_with_callback_url
+  end
+
+  def test_poll_destination_powered_off_in_vmdb
+    @test_poll_destination_powered_off_in_vmdb_setup ||= begin
```

Comment:
States can be requeued which calls them multiple times.  We only want to do the setup the first time it is called.